### PR TITLE
Improve command to change user and group from version 4.2.x to 4.x.x

### PR DESCRIPTION
**Pull request by GabrielEValenzuela**
|Related issue|
|---|
| #14175  |

# Solution report

## Goal
This report has for goal to make a summary of the implemented solution.

## Tests description
To solve this problem, two VMs with 2 cores and 1Gb of RAM, both from CentOS, were used, one locally and one being an AWS instance. First installed a manager version 4.2.7 in both VMs and take the time neccesary to update using the old version of installer, i.e that one with the command `find ./folder -user u -group g -exec chown nu:ng {} \;` and using the new version, i.e, with the command `find ./folder -user u -group g -print0 | xargs -0 chown nu:ng`.

In both VMs a half million of files was created on the folder `/var/ossec/quque/rids`

## Results
For the local VM, the following results were got:

| update w/ | time[s]   |
|--------------|---------|
| old version | 566.92 |
| new version | 38.87 |
_(Table 1: Local VM with 2 core and 1Gb of memory. Update from 4.2.7 to 4.x.x)_

Also, we can see the following proof that te command take almost the 90% of the update process.

| update w/ | time[s]   |
|--------------|---------|
| old version | 965.01 |
| new version | 14.57 |
_(Table 2: AWS VM with 2 core and 1Gb of memory. Update from 4.2.7 to 4.x.x)_

long_command_proof
_(Figure 1: Proof command long execution time on local VM)_

command_aws
_(Figure 2: Proof command long execution time on AWS VM)_

split_command_chown
_(Figure 3: Proof aplit the new chown command in `find` and `xargs`)_

## Conclutions
The improvement show an increment of speed of 4000% in average, proving that the use of `xargs` it's the best option to solve the issue.

- Compilation without warnings in every supported platform
  - [x] Linux
  - [ ] Windows
  - [ ] MAC OS X
- [x] Source installation
- [x] Package installation
- [ ] Source upgrade
- [x] Package upgrade
- [x] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Valgrind (memcheck and descriptor leaks check)
  - [ ] Dr. Memory
  - [ ] AddressSanitizer
- Memory tests for Windows
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Dr. Memory
- Memory tests for macOS
  - [ ] Scan-build report
  - [ ] Leaks
  - [ ] AddressSanitizer

- [ ] Retrocompatibility with older Wazuh versions
- [ ] Working on cluster environments
- [ ] Configuration on demand reports new parameters
- [ ] The data flow works as expected (agent-manager-api-app)
- [ ] Added unit tests (for new features)
- [ ] Stress test for affected components

- Decoder/Rule tests
  - [ ] Added unit testing files ".ini"
  - [ ] runtests.py executed without errors

### CODE REVIEW

File: src/init/wazuh/wazuh.sh [hunk 1]
```diff
@@ -209,8 +209,8 @@ WazuhUpgrade()
     # Replace and delete ossec group along with ossec users
     OSSEC_GROUP=ossec
     if (grep "^ossec:" /etc/group > /dev/null 2>&1) || (dscl . -read /Groups/ossec > /dev/null 2>&1)  ; then
-        find $PREINSTALLEDDIR -group $OSSEC_GROUP -user root -exec chown root:wazuh {} \;
-        find $PREINSTALLEDDIR -group $OSSEC_GROUP -exec chown wazuh:wazuh {} \;
+        find $PREINSTALLEDDIR -group $OSSEC_GROUP -user root -print0 | xargs -0 root:wazuh
```
Comment (chemamartinez) [inline src/init/wazuh/wazuh.sh:212]: Missing the `chown` command.
```suggestion
        find $PREINSTALLEDDIR -group $OSSEC_GROUP -user root -print0 | xargs -0 chown root:wazuh
```

File: src/init/wazuh/wazuh.sh [hunk 2]
```diff
@@ -209,8 +209,8 @@ WazuhUpgrade()
...
+        find $PREINSTALLEDDIR -group $OSSEC_GROUP -print0 | xargs -0 wazuh:wazuh
```
Comment (chemamartinez) [inline src/init/wazuh/wazuh.sh:213]: Same

**Review by chemamartinez — CHANGES_REQUESTED** [review]

File: src/init/wazuh/wazuh.sh (hunk 1 above)
Comment (GabrielEValenzuela) [reply src/init/wazuh/wazuh.sh]: Fix it !

File: src/init/wazuh/wazuh.sh (hunk 2 above)
Comment (GabrielEValenzuela) [reply src/init/wazuh/wazuh.sh]: Fix it !

**Review by chemamartinez — DISMISSED** [review]
LGTM!

**Review by jmv74211 — DISMISSED** [review]
## QA review

- **Type**: Manual testing.
- **Status**: Approved  🟢 
- **Testing issue**: https://github.com/wazuh/wazuh-qa/issues/3745
- **Comments**:  https://github.com/wazuh/wazuh-qa/issues/3745#issuecomment-1397081826

**Review by chemamartinez — APPROVED** [review]

**Pull request merged**